app/models/manageiq/providers/openstack/cloud_manager/provision/cloning.rb

```diff
@@ -1,8 +1,13 @@
 module ManageIQ::Providers::Openstack::CloudManager::Provision::Cloning
   def do_clone_task_check(clone_task_ref)
-    source.with_provider_connection do |openstack|
-      instance = openstack.handled_list(:servers).detect { |s| s.id == clone_task_ref }
-      status   = instance.state.downcase.to_sym
+    connection_options = {:tenant_name => options[:cloud_tenant][1]} if options[:cloud_tenant].kind_of?(Array)
+    source.with_provider_connection(connection_options) do |openstack|
+      instance = if connection_options
+                   openstack.servers.get(clone_task_ref)
+                 else
+                   openstack.handled_list(:servers).detect { |s| s.id == clone_task_ref }
+                 end
+      status   = instance.state.downcase.to_sym if instance.present?
```

Comment:
Thanks for the PR. I think we should initialize ```status``` local variable always to avoid _undefined variable error_ since it is part of condition on line 12.

Reply:
`status   = instance.state.downcase.to_sym if instance.present?` results in `status` being defined as `nil` if the condition is false, so an additional initialization shouldn't be necessary.
```
irb(main):001:0> x = 3 if true
=> 3
irb(main):002:0> x
=> 3
irb(main):003:0> y = 3 if false
=> nil
irb(main):004:0> y
=> nil
irb(main):005:0> z
NameError: undefined local variable or method `z' for main:Object
```

Reply:
Ah, thanks!